reconcile/test/test_utils_ldap_client.py

```diff
@@ -0,0 +1,43 @@
+import ldap3
+import pytest
+from ldap3 import Connection
+from reconcile.utils.ldap_client import LdapClient
+
+
+@pytest.fixture
+def connection_search_result():
+    result = [
+        {"attributes": {"uid": ["user1"]}},
+        {"attributes": {"uid": ["user2"]}},
+        {"attributes": {"uid": ["user3"]}},
+    ]
+    return result
+
+
+def test_ldap_client_from_settings(mocker, connection_search_result):
+    mock_connection_bind = mocker.patch.object(Connection, "bind", autospec=True)
+    mock_connection_unbind = mocker.patch.object(Connection, "unbind", autospec=True)
+    mock_connection_search = mocker.patch.object(Connection, "search", autospec=True)
+    mock_connection_search.return_value = None, None, connection_search_result, None
+
+    settings = {"ldap": {"baseDn": "test", "serverUrl": "testUrl"}}
+    with LdapClient.from_settings(settings) as ldap_client:
+        uids = ["user1", "user2", "user3"]
+        ldap_client.get_users(uids)
+
+    mock_connection_search.assert_called_once()
+    mock_connection_bind.assert_called_once()
+    mock_connection_unbind.assert_called_once()
```

Comment:
And then these don't seem to be taking any arguments, so:
```suggestion
    mock_connection_bind.assert_called_once_with()
    mock_connection_unbind.assert_called_once_with()
```